src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,101 @@
+   */
+  options: {
+    label: string;
+    value: string;
+  }[];
+  /**
+   * Current selected value
+   */
+  value: DropdownSelect.SelectProps['defaultValue'];
+  /**
+   * Pass a call back that triggers once a new value is selected
+   */
+  onValueChange: (value: string) => void;
+  /**
+   * Aria-label for specific dropdown use
+   */
+  label: string;
+}
+
+/**
+ * ## Overview
+ *
+ * Overview of this component
+ *
+ * [Figma Link](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=10570-6295&t=EnYSS6Hk58EiWaVg-4)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-dropdown--overview)
+ */
+const Dropdown = React.forwardRef<HTMLDivElement, DropdownProps>(
+  ({ options, value, onValueChange, label, className, id, ...props }, ref) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps({ id }, 'Dropdown');
+    const [isOpen, setIsOpen] = useState(false);
+
+    return (
+      <div className={classnames(baseClassName, className)} id={id} {...commonProps} {...props} ref={ref}>
+        <DropdownSelect.Root
+          defaultValue={value}
+          onValueChange={(value) => {
+            onValueChange(value);
+          }}
+          onOpenChange={setIsOpen}
+        >
+          <DropdownSelect.Trigger className={classnames(`${baseClassName}__trigger`)} aria-label={label}>
+            <DropdownSelect.Value placeholder={value} />
+            <DropdownSelect.Icon>{isOpen ? <ChevronUpIcon /> : <ChevronDownIcon />}</DropdownSelect.Icon>
+          </DropdownSelect.Trigger>
+          <DropdownSelect.Portal>
+            <DropdownSelect.Content className={classnames(`${baseClassName}__content`)} position="popper">
+              <DropdownSelect.ScrollUpButton className={classnames(`${baseClassName}__scroll-button`)}>
+                <ChevronUpIcon />
+              </DropdownSelect.ScrollUpButton>
+              <DropdownSelect.Viewport className={classnames(`${baseClassName}__viewport`)}>
+                {options.map((option) => {
+                  return (
+                    <DropdownItem
+                      key={option.value}
+                      value={option.value}
+                      disabled={option.value === value}
+                      className={baseClassName}
+                    >
+                      {option.label}
+                    </DropdownItem>
+                  );
+                })}
+              </DropdownSelect.Viewport>
+              <DropdownSelect.ScrollDownButton className={classnames(`${baseClassName}__scroll-button`)}>
+                <ChevronDownIcon />
+              </DropdownSelect.ScrollDownButton>
+            </DropdownSelect.Content>
+          </DropdownSelect.Portal>
+        </DropdownSelect.Root>
+      </div>
+    );
+  },
+);
+
+Dropdown.displayName = 'Dropdown';
+
+const DropdownItem = React.forwardRef<HTMLDivElement, DropdownSelect.SelectItemProps>(
```

Comment:
Do we need this to be a forwarded ref? We are not passing any refs to this component. Does Radix do some magic to get this ref passed?

Reply:
The main implementation examples had the `DropdownItem` set up like this, so I followed their lead: https://www.radix-ui.com/primitives/docs/components/select

I can remove for now if we want, but I'm not positive of what magic is around it

Reply:
Since focus is fully managed they probably use this ref internally.

Reply:
@davidicus this was my idea just in case some consumer needed to measure the width or really call any Javascript imperative function on the element. (is the element in the viewport, etc)

Reply:
Yeah if that is the case we have to be able to pass the ref from props, no?